### DISCUSSION TITLE
Propagating execution date in scheduled dataset DAGs

### DISCUSSION THREAD

**Discussion by mullenpaul** [apache/airflow#37450] [Q&A]
Hi all,

We have a reasonably complex workflow which we have decomposed into multiple DAGs. Each of these DAGs have a defined output and the DAGs are dependent on each other. As these pipelines update we want to be able to backfill several years worth of data.

Currently the DAGs are just have their schedule offset which we know isn't great and are looking for a better solution. From what I understand the current solution would be to have an upstream DAG trigger a downstream DAG via `TriggerDagRunOperator` which would allow us to pass the logical execution date forward - so that say 10 Nov 2020 would be used as the execution date in the downstream DAG.

I'm not a huge fan of this kind of approach as it explicitly ties DAGs together rather than mapping a Dataset as an output of a DAG which others can listen to. It would be great to use Datasets to sync downstream DAGs for backfilling.

My question is: this doesn't seem to be something capable with Datasets currently - have I got this approach wrong or is there another way to achieve what we want?

If this isn't the case and this is something not supported - with Datasets - would a PR to this effect be welcome?

Cheers.

**Comment by potiuk** [comment]
Yes. Datasets is your solution

**Reply by mullenpaul** [reply]
Do you have an idea as to how we can have the execution date passed between DAGs? Ideally, the downstream DAG should be able to have its execution date set to what the upstream DAG has.

**Reply by potiuk** [reply]
Read the datasets docs.

**Reply by potiuk** [reply]
And just to add - just try and see. I have no detailed answers to all the questions as I have not developed datasets - but you can just try and see how it works.

**Reply by potiuk** [reply]
Actually, liekly the state of what you are looking for is nicely described and some proposals of improvements here: https://github.com/apache/airflow/discussions/34050#discussioncomment-8482046  so likely you should take a look there (and maybe even take part in the discussion)
